src/backend/distributed/commands/utility_hook.c

```diff
@@ -276,6 +277,22 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 				UndistributeDisconnectedCitusLocalTables();
 			}
 			ResetConstraintDropped();
+
+			if (context == PROCESS_UTILITY_TOPLEVEL &&
+				NeedAddToMetadataIfTableCreated(parsetree))
```

Comment:
Maybe ShouldAddNewTableToMetadata

mainly, we usually start these function names with Should